Configurations/00-base-templates.conf

```diff
@@ -116,18 +116,18 @@ my %targets=(
                 return ();
             },
 
-        ld              => "link",
-        lflags          => "/nologo",
-        loutflag        => "/out:",
-        ar              => "lib",
-        arflags         => "/nologo",
-        aroutflag       => "/out:",
-        rc               => "rc",
-        rcoutflag        => "/fo",
-        mt              => "mt",
-        mtflags         => "-nologo",
-        mtinflag        => "-manifest ",
-        mtoutflag       => "-outputresource:",
+        LD              => "link",
+        LDFLAGS         => "/nologo",
+        LDOUTFLAG       => "/out:",
+        AR              => "lib",
+        ARFLAGS         => "/nologo",
+        AROUTFLAG       => "/out:",
+        RC               => "rc",
+        RCOUTFLAG        => "/fo",
+        MT              => "mt",
+        MTFLAGS         => "-nologo",
+        MTINFLAG        => "-manifest ",
+        MTOUTFLAG       => "-outputresource:",
```

Comment:
On a **side** note, just for the sake of speaking my mind, \*OUT\* and MT [and eve RC] is the kind of parameterization I find a bit excessive. I'd say they can as well be just coded into windows-makefile.template. Yes, one can argue that there might be some other non-Microsoft toolchain when these would be called something else, but can we actually say that? Template is in fact nmake[!] specific. One can probably argue that we could just as well call it vc-makefile.tmpl. Once again, **side** note, just forget about it :-)

Reply:
We've had that discussion before, and while I agree that the template is *nmake* *syntax* specific, I do not agree that it's *vc* specific.

Reply:
... or it's at least not meant to be *vc* specific...  there may be details still that make this statement not entirely true, but the intention is there.

Reply:
Rationale basically is that nmake is something that there is available only with VC, which makes whole thing VC-specific. But of course I can appreciate that templates (or rather unix-Makefile one) are not meant to be tool-chain specific and that it helps to view them *all* as such. However, it's not given that you can escape platform- or tool-chain-specific *rules*. For example, there are mingw-specific rules that use windres in unix-Makefile.template. Currently they disguise as filename-extension-specific, but it's only possible because extension is not used on some other platform. In other words what I'm saying is that I for one wouldn't "feel bad" about having some rules rigid, with no parameterization knobs, sometimes it's appropriate...

Reply:
I so strongly disagree with you.  Basically, what you're saying is that if you use nmake, you can only use it with VC.  That's like saying that you only can use GNU make with the GNU C toolchain.

My intention was always that the different templates would be compatible with a particular 'make' implementation for each "platform", but never tied to a particular toolchain (on a best effort basis).  I don't want to change that.

Reply:
> what you're saying is that if you use nmake, you can only use it with VC.

Trick question... Let's put it this way. If you are using VC, nmake is effectively the only option. [Just in case, using icc for Windows is equivalent to using VC, because it relies *heavily* on remaining infrastructure components.] Other Windows tool-chains do come with own make utilities, which are customarily GNU make. I'm not exactly saying that if you use nmake, you can only use it with VC, but rather that there is nothing else you can use it with. This is just a de-facto statement. But then I also said that I can appreciate that all templates are viewed as tool-chain independent, it *is* useful abstraction. And I also shifted focus toward platform-specific rules. In other words I'm cool with nmake being viewed as tool-chain independent, but it still doesn't have to mean that one has to parameterize down to say MTINFLAG and MTOUTFLAGS. And suggested alternative is to express things like manifest embedding as platform- or tool-chain-specific rules in otherwise tool-chain-neutral template.